### ISSUE TITLE
DELETE posted JARS

### ISSUE THREAD

**Issue by jpoblete** [spark-jobserver/spark-jobserver#454]
Hello,

We are using your version 0.5.2 where this code allow to GET/POST JAR files

https://github.com/spark-jobserver/spark-jobserver/blob/v0.5.2/job-server/src/spark.jobserver/WebApi.scala#L56

```
  /**
   * Routes for listing and uploading jars
   *    GET /jars              - lists all current jars
   *    POST /jars/<appName>   - upload a new jar file
   */
```

Does not seem like a DELETE /jars/<appname> is implemented
I looked if that would be available on the latest version but it seems only GET/POST/DELETE of datafiles is supported.

Could you clarify if the current version would allow for GET/POST/DELETE of JAR files
OR explain what options we have?

Thanks a lot!
-JMP

**Comment by velvia** [comment]
Only GET and POST of /jars is supported, and this has not changed since 0.5.2.   DELETE would not be hard though to implement.  Should we turn this issue into a DELETE feature?

**Comment by hntd187** [comment]
I think it'd be easy to add it, but we should prolly make it an opt-in feature, because in the wrong configuration it could be dangerous.

**Comment by velvia** [comment]
+1 for opt-in

**Comment by jpoblete** [comment]
As it is right now, the job server creates a new jar every time a jar is curled, even when it has the exact same name, it just adds the timestamp to the end like this:

```
-rw-rw-r--+ 1 cassandra cassandra 49353913 May 9 14:35 ihub-cwe-extract-2016-05-09T14_35_27.061-04_00.jar

-rw-rw-r--+ 1 cassandra cassandra 49353913 May 9 14:49 ihub-cwe-extract-2016-05-09T18_49_12.674Z.jar

-rw-rw-r--+ 1 cassandra cassandra 49353913 May 9 15:59 ihub-cwe-extract-2016-05-09T15_59_23.681-04_00.jar
```

For now DELETE would have to be a manual maintenance task

**Comment by mattweyant** [comment]
+1

**Comment by velvia** [comment]
Only GET and POST of /jars is supported, and this has not changed since
0.5.2.   DELETE would not be hard though to implement.  Should we turn this
issue into a DELETE feature?

On May 5, 2016, at 12:35 PM, José Martínez Poblete <notifications@github.com>
wrote:

Hello,

We are using your version 0.5.2 where this code allow to GET/POST JAR files

https://github.com/spark-jobserver/spark-jobserver/blob/v0.5.2/job-server/src/spark.jobserver/WebApi.scala#L56

  /**
   * Routes for listing and uploading jars
   *    GET /jars              - lists all current jars
   *    POST /jars/<appName>   - upload a new jar file
   */

Does not seem like a DELETE /jars/ is implemented
I looked if that would be available on the latest version but it seems only
GET/POST/DELETE of datafiles is supported.

Could you clarify if the current version would allow for GET/POST/DELETE of
JAR files
OR explain what options we have?

Thanks a lot!
-JMP

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub
<https://github.com/spark-jobserver/spark-jobserver/issues/454>

**Comment by f1yegor** [comment]
PR #598

**Comment by velvia** [comment]
Nice, thanks

-Evan
"Never doubt that a small group of thoughtful, committed citizens can change the world" - M. Mead

**Comment by doernemt** [comment]
Fixed with PR #623, part of 0.7 release. @velvia

**Comment by noorul** [comment]
Yes indeed! Closing.
